flashinfer/fused_moe/layer.py

```
@@ -94,19 +95,20 @@ def __init__(self, config: MoEConfig, device: Optional[torch.device] = None):
             runner_cls = _BACKEND_RUNNERS.get(type(backend_cfg))
             if runner_cls is None:
                 continue  # MVP scope — skip non-MVP backends silently
-            # Skip backends that cannot execute the configured quant variant, so a
-            # mixed candidate list (e.g. BF16 with (CuteDslConfig, TrtllmBf16Config))
-            # never instantiates a runner that would mis-handle the pack contract.
-            if config.quant.variant not in _RUNNER_QUANTS.get(type(backend_cfg), ()):
+            runner = runner_cls(config, device=self.device)
+            try:
+                runner.check_support()
+            except Exception:
                 continue
```

Ruff (0.15.21)

[error] `try`-`except`-`continue` detected, consider logging the exception

(S112)